paddle/fluid/operators/benchmark/op_tester.cc

```diff
@@ -0,0 +1,350 @@
+      var->SetType(framework::proto::VarType::LOD_TENSOR);
+      var->SetPersistable(false);
+      var->SetDataType(framework::proto::VarType::FP32);
+
+      op_desc_.SetOutput(name, {var_name});
+      outputs_.push_back(var_name);
+    }
+  } else {
+    LOG(FATAL) << "Op \"" << type << "\" is not registered.";
+  }
+
+  if (/* use_gpu */ false) {
+    place_ = paddle::platform::CUDAPlace(0);
+  } else {
+    place_ = paddle::platform::CPUPlace();
+  }
+
+  framework::InitDevices(false);
+  scope_.reset(new paddle::framework::Scope());
+
+  op_ = framework::OpRegistry::CreateOp(op_desc_);
+  CreateVariables(scope_.get());
+}
+
+void OpTester::Init(const std::string &filename) {
+  config_ = OpTesterConfig(filename);
+
+  auto &op_desc_info = framework::OpInfoMap::Instance();
+  // Initialize the OpDesc
+  if (op_desc_info.Has(config_.op_type)) {
+    type_ = config_.op_type;
+    op_desc_.SetType(config_.op_type);
+    std::vector<std::string> input_names = GetOpProtoInputNames();
+    for (auto &name : input_names) {
+      const OpInputConfig *input = config_.GetInput(name);
+      if (input == nullptr) {
+        LOG(FATAL) << "Donot correctlly provide input " << name;
+      }
+
+      std::string var_name = config_.op_type + "." + name;
+      framework::VarDesc *var = Var(var_name);
+      // Need to support more type
+      var->SetType(framework::proto::VarType::LOD_TENSOR);
+      var->SetPersistable(false);
+      var->SetDataType(framework::proto::VarType::FP32);
+      var->SetShape(input->dims);
+
+      op_desc_.SetInput(name, {var_name});
+      inputs_.push_back(var_name);
+    }
+    std::vector<std::string> output_names = GetOpProtoOutputNames();
+    for (auto &name : output_names) {
+      std::string var_name = config_.op_type + "." + name;
+      framework::VarDesc *var = Var(var_name);
+      // Need to support more type
+      var->SetType(framework::proto::VarType::LOD_TENSOR);
+      var->SetPersistable(false);
+      var->SetDataType(framework::proto::VarType::FP32);
+
+      op_desc_.SetOutput(name, {var_name});
+      outputs_.push_back(var_name);
+    }
+  } else {
+    LOG(FATAL) << "Op \"" << config_.op_type << "\" is not registered.";
+  }
+
+  if (config_.use_gpu) {
+    place_ = paddle::platform::CUDAPlace(0);
+  } else {
+    place_ = paddle::platform::CPUPlace();
+  }
+
+  framework::InitDevices(false);
+  scope_.reset(new paddle::framework::Scope());
+
+  op_ = framework::OpRegistry::CreateOp(op_desc_);
+  CreateVariables(scope_.get());
+}
+
+void OpTester::Run() {
+  if (config_.print_debug_string) {
+    LOG(INFO) << DebugString();
+  }
+
+  // Warm up
+  RunImpl();
+
+  if (config_.profile) {
+    if (platform::is_cpu_place(place_)) {
+      platform::EnableProfiler(platform::ProfilerState::kCPU);
+    } else {
+#ifdef PADDLE_WITH_CUDA
+      platform::EnableProfiler(platform::ProfilerState::kAll);
+      // The default device_id of paddle::platform::CUDAPlace is 0.
```

Comment:
This comments cloud be options of config, or just remove.

Reply:
Done.